tests/unit/s2n_resume_test.c

```diff
@@ -338,5 +343,80 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
     }
+
+    /* s2n_config_set_session_ticket_callback */
+    {
+        struct s2n_config *config = NULL;
+        EXPECT_NOT_NULL(config = s2n_config_new());
+
+        /* Safety check */
+        {
+            EXPECT_FAILURE_WITH_ERRNO(s2n_config_set_session_ticket_callback(NULL, s2n_test_session_ticket_callback), S2N_ERR_NULL);
+        }
+
+        EXPECT_NULL(config->session_ticket_cb);
+        EXPECT_SUCCESS(s2n_config_set_session_ticket_callback(config, s2n_test_session_ticket_callback));
+        EXPECT_EQUAL(config->session_ticket_cb, s2n_test_session_ticket_callback);
+        EXPECT_SUCCESS(s2n_config_free(config));
+    }
+
+    /* s2n_session_ticket_get_data */
+    {
+        /* Safety checks */
+        {
+            struct s2n_session_ticket session_ticket = { 0 };
+            uint8_t *data = NULL;
+            size_t data_len = 0;
+            EXPECT_FAILURE_WITH_ERRNO(s2n_session_ticket_get_data(NULL, &data, &data_len), S2N_ERR_NULL);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_session_ticket_get_data(&session_ticket, NULL, &data_len), S2N_ERR_NULL);
+            EXPECT_FAILURE_WITH_ERRNO(s2n_session_ticket_get_data(&session_ticket, &data, NULL), S2N_ERR_NULL);
+        }
+
+        /* Empty ticket */
+        {
+            struct s2n_session_ticket session_ticket = { 0 };
+            uint8_t *data = NULL;
+            size_t data_len = 0;
+            EXPECT_SUCCESS(s2n_session_ticket_get_data(&session_ticket, &data, &data_len));
+            EXPECT_EQUAL(data_len, 0);
+            EXPECT_EQUAL(data, NULL);
+        }
+
+        /* Valid ticket */
+        {
+            uint8_t ticket_data[] = "session ticket data";
+            struct s2n_blob ticket_blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&ticket_blob, ticket_data, sizeof(ticket_data)));
+            struct s2n_session_ticket session_ticket = { .ticket_data = ticket_blob };
+
+            uint8_t *data = NULL;
+            size_t data_len = 0;
+            EXPECT_SUCCESS(s2n_session_ticket_get_data(&session_ticket, &data, &data_len));
+            EXPECT_EQUAL(data_len, sizeof(ticket_data));
```

Comment:
Shouldn't this be `sizeof(ticket_data)/sizeof(uint8_t)`? Isn't data_len the length of the ticket data?

Reply:
Isn't sizeof(uint8_t) = 1? I'm not sure what you mean by this comment.

Reply:
The concern was for different/older systems, can we safely assume that sizeof(uint8_t) == 1 for all systems?

Reply:
Yes. The Point of the "uintX_t" types is that they are exactly a certain size. If that was not true, we'd have more to worry about than our tests being wrong :)

Reply:
This stack overflow [question](https://stackoverflow.com/questions/48655310/can-i-assume-that-sizeofuint8-t-1) answers the above question.